pkgs/build-support/kernel/modules-closure.sh

```diff
@@ -80,7 +80,7 @@ for module in $(< ~-/closure); do
     # of its output.
     modinfo -b $kernel --set-version "$version" -F firmware $module | grep -v '^name:' | while read -r i; do
         echo "firmware for $module: $i"
-        for name in "$i" "$i.xz" ""; do
+        for name in "$i" "$i.xz" "$i.zst" ""; do
             [ -z "$name" ] && echo "WARNING: missing firmware $i for module $module"
             if cp -v --parents --no-preserve=mode lib/firmware/$name "$out" 2>/dev/null; then
```

Comment:
I know this is unrelated to the PR, but this is such a weird way to write it.
I wonder why it's not someting like:
```bash
if ! cp -v --parents --no-preserve=mode lib/firmware/$i{,.xz,.zst} "$out" 2>/dev/null; then
    echo "WARNING: missing firmware $i for module $module"
fi
```

Reply:
That will copy more than one file (though don't know if that can be the case here) and also I think it's really unintuitive here to rely on cp's behavior to keep going if one of the source files doesn't exist (I had to check if it actually does that). Sure, you could write `for name in "$i"{,.xz,.zst}` still, but meh. Above code looks fine to me except for [ instead of [[ (and that's not really something critical).